src/operator/nn/mkldnn/mkldnn_base-inl.h

```diff
@@ -327,7 +327,8 @@ typedef std::pair<OutDataOp, mkldnn::memory *> mkldnn_output_t;
  * If these two functions are used, we have to call CommitOutput to write
  * the output back to the output NDArray.
  */
-mkldnn_output_t CreateMKLDNNMem(const NDArray &arr,
+mkldnn_output_t CreateMKLDNNMem(const NDArray &out_arr,
```

Comment:
can you create a separate function for this? so we don't need to modify all mkldnn operators.

Reply:
I thought about that, but then technically wouldn't all shouldn't all operators be eligible to use WriteInPlace?

Reply:
+1 for overriding this function.

Reply:
@azai91 not really. WriteInplace only works in a few operators (sum, activation, batchnorm). It doesn't work on some operators such as convolution and pooling for sure because the inputs and outputs of these operators don't have the same shape. 

Ideally, we should have two functions for this, so one of them have an interface consistent with `CreateMKLDNNWeightGrad` (creating MKLDNN memory from any NDArray, instead of just output NDArrays).

If you prefer to use one function, please update the comment and explain why `CreateMKLDNNMem` needs `in_arr`. Meanwhile, please rename `arr` of `CreateMKLDNNWeightGrad ` to `out_arr`, so `CreateMKLDNNMem` and `CreateMKLDNNWeightGrad` are still consistent.